common/iso_config.go

```diff
@@ -37,50 +35,36 @@ import (
 //
 // ```json
 // {
-//   "iso_checksum_type": "file",
-//   "iso_checksum": "ubuntu.org/..../ubuntu-14.04.1-server-amd64.iso.sum",
+//   "iso_checksum": "file:ubuntu.org/..../ubuntu-14.04.1-server-amd64.iso.sum",
 //   "iso_url": "ubuntu.org/.../ubuntu-14.04.1-server-amd64.iso"
 // }
 // ```
 //
 // ```json
 // {
-//   "iso_checksum_url": "./shasums.txt",
+//   "iso_checksum": "file://./shasums.txt",
 //   "iso_url": "ubuntu.org/.../ubuntu-14.04.1-server-amd64.iso"
 // }
 // ```
 //
 // ```json
 // {
-//   "iso_checksum_type": "sha256",
-//   "iso_checksum_url": "./shasums.txt",
+//   "iso_checksum": "file:./shasums.txt",
```

Comment:
Nope, because the prefix is `file` we will try to get it first

Reply:
But reading the file packer will first check if the line contains a hashing algo but if it doesnt then yes it will try to guess it based on the length.